hibernate-reactive-core/src/main/java/org/hibernate/reactive/persister/collection/impl/ReactiveBasicCollectionPersister.java

```diff
@@ -222,11 +228,16 @@ protected String getSQLUpdateRowString() {
 		return CompletionStages.total(
 				collection.entries( this ),
 				entry -> {
-					CompletionStage<Integer> st = reactiveConnection.update(
-							getSQLInsertRowString(),
-							insertRowsParamValues( entry, index, collection, id, session ) );
-					collection.afterRowInsert( this, entry, index.getAndIncrement() );
-					return st;
+					if ( collection.needsInserting( entry, index.get(), elementType ) ) {
```

Comment:
I think you need something like this here:
```
int i = index.getAndIncrement();
if ( collection.needsInserting( entry, i, elementType ) ) {
...
}
```
Or you only increment the index when the collection needs inserting.
This behaviour would be different than what happens in ORM: https://github.com/hibernate/hibernate-orm/blob/f451ed46d9d2856f9dab6a4ebb985a4d3ea8fee5/hibernate-core/src/main/java/org/hibernate/persister/collection/AbstractCollectionPersister.java#L1614

Reply:
I suspect there are no failures now because we don't have a test using `@Embeddable`

Reply:
... right. In ORM version, the index is incremented whether or not entry `needsInserting(entry...)` so I'm cleaning that up. Note that `i` is 0 based, so the incrementing can't happen before the check.. putting it outside of the check.

Reply:
```
int i = index.getAndIncrement();
```
This method will first return the value **and then** increment it.
Meaning the `i` will start from 0.
If this is what you mean, because index actually start from 1 :-)

Reply:
You won't use index anymore after this, so it should be correct

Reply:
dah.. yep.. thx

Reply:
Which @Embeddable scenario are you referring to?  

this one:
```
		@ElementCollection(fetch = FetchType.EAGER)
		private Map<String, Phone> phones = new HashMap<>();
```
or the @Embeddable with it's own @ElementCollection?

Reply:
Yes, I mean this case:
```
@ElementCollection(fetch = FetchType.EAGER)
private Map<String, Phone> phones = new HashMap<>();
```

Reply:
done